controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -845,11 +852,16 @@ func (r *BareMetalHostReconciler) actionProvisioning(prov provisioner.Provisione
 	}
 
 	// If the provisioner had no work, ensure the image settings match.
-	if info.host.Status.Provisioning.Image != *(info.host.Spec.Image) {
+	if info.host.Spec.Image != nil && info.host.Status.Provisioning.Image != *(info.host.Spec.Image) {
 		info.log.Info("updating deployed image in status")
 		info.host.Status.Provisioning.Image = *(info.host.Spec.Image)
 	}
 
+	if info.host.Spec.CustomDeploy != nil && (info.host.Status.Provisioning.CustomDeploy == nil || *(info.host.Spec.CustomDeploy) != *(info.host.Status.Provisioning.CustomDeploy)) {
```

Comment:
That last comparison is I think valid because the struct only contains a string, but if it contained anything more complex (and the whole point of defining a struct is to give us an option of adding more complex stuff in future AIUI) then it wouldn't. If we use reflect.DeepEqual() here now we can save someone a headache in future. (Yes, the image comparison above is also dodgy.)